### PR TITLE
@nofrixion/moneymoov Npm Package

### PR DESCRIPTION
**Pull request by donalnofrixion**
First version of the @nofrixion/moneymoov npm package.

Contains api clients, hooks and types for the MoneyMoov api.

TODO:
Port over custom hooks and types from the nofrixion.business repo.

### CODE REVIEW

**Review by arifmatin — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -1,2 +1,58 @@
...
+- [Visual Studio Code](https://code.visualstudio.com/)
+  - [ESLint](https://marketplace.visualstudio.com/items?itemName=dbaeumer.vscode-eslint)
+  - [Prettier](https://marketplace.visualstudio.com/items?itemName=esbenp.prettier-vscode)
+
+## 🚀 Getting Started
+
+1. Clone the repository
+
+```bash
+git clone git@github.com:nofrixion/nofrixion.webcomponents.git
```
Comment (Publicker) [inline README.md]: Wrong git clone path

File: src/clients/BaseApiClient.ts [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+   */
+  protected async httpRequest<TResponse>(
+    url: string,
+    method: HttpMethod,
+    postData?: unknown,
+  ): Promise<{
+    data?: TResponse
+    error?: ApiError
+  }> {
+    console.log(`Requesting: ${method} ${url}`)
```
Comment (Publicker) [inline src/clients/BaseApiClient.ts]: What do you think of adding a new param to the `BaseApiClient` constructor like `debug`, and if that comes true, console logs will show up, but won't if that comes as false, and the default value being false.

File: src/responseTypes/ApiResponses.ts
Comment (Publicker) [inline src/responseTypes/ApiResponses.ts:1]: I believe this file also contains ApiRequests. Can we check that? If that's the case, can we split it into different files?

File: src/clients/BaseApiClient.ts (hunk 2 above)
Comment (donalnofrixion) [reply src/clients/BaseApiClient.ts]: Sounds good.

File: src/clients/BaseApiClient.ts [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+      const error = ex as AxiosError
+
+      if (error.response?.status === 401) {
+        // Unauthorized
+        this.onUnauthorized()
+      }
+
+      if (error.response?.data) {
+        // This contains the problem details
+        console.log('Received error from api. : ' + JSON.stringify(error.response?.data))
```
Comment (Publicker) [inline src/clients/BaseApiClient.ts]: Shall we enclose this `console.log` inside a `if (debug)`?

**Review by Publicker — APPROVED** [review]

**Pull request merged**